local controllers = script.Parent:WaitForChild("controllers")

local PlacementController = require(controllers.PlacementController)
local UIController        = require(controllers.UIController)
local HUDController       = require(controllers.HUDController)
local EnemyController     = require(controllers.EnemyController)
local TowerController     = require(controllers.TowerController)
local WeaponController    = require(controllers.WeaponController)
local StoreController     = require(controllers.StoreController)
local SellController               = require(controllers.SellController)
local WeaponShopController         = require(controllers.WeaponShopController)
local WeaponInventoryController    = require(controllers.WeaponInventoryController)
local WeaponNotificationController = require(controllers.WeaponNotificationController)

-- Cada controller roda independente: um travado não bloqueia os outros
task.spawn(function() PlacementController:init() end)
task.spawn(function() UIController:init() end)
task.spawn(function() HUDController:init() end)
task.spawn(function() EnemyController:init() end)
task.spawn(function() TowerController:init() end)
task.spawn(function() WeaponController:init() end)
task.spawn(function() StoreController:init() end)
task.spawn(function() SellController:init() end)
task.spawn(function() WeaponShopController:init() end)
task.spawn(function() WeaponInventoryController:init() end)
task.spawn(function() WeaponNotificationController:init() end)
